06_puppy/runnerdave/map_store.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import "fmt"
+
+// MapStore represents a simple map storage for the Puppy store
+type MapStore struct {
+	puppies map[uint16]Puppy
+}
+
+// NewMapStore creates a new in-memory store with map intialised
+func NewMapStore() *MapStore {
+	var m MapStore
+	m.puppies = make(map[uint16]Puppy)
+	return &m
+}
+
+// CreatePuppy saves new puppy if not in store, if it is already returns error
+func (m *MapStore) CreatePuppy(p *Puppy) error {
```

Comment:
Since your store stores `Puppy` rather than `*Puppy`, I think there is no need to accept `*Puppy` as argument here.

Reply:
Good point I have made a change so my store stores *Puppy now